src/client/components/pages/NonClassMeetings/NonClassMeetingsTable.tsx

```diff
@@ -0,0 +1,244 @@
+import React, {
+  FunctionComponent,
+  ReactElement,
+} from 'react';
+import {
+  Table,
+  TableHead,
+  TableBody,
+  TableRow,
+  TableHeadingCell,
+  TableCell,
+  TableHeadingSpacer,
+  TableCellList,
+  TableCellListItem,
+  BorderlessButton,
+  VARIANT,
+  fromTheme,
+} from 'mark-one';
+import NonClassMeetingResponseDTO from 'common/dto/nonClassMeetings/NonClassMeeting.dto';
+import { getAreaColor } from 'common/constants';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faEdit, faFolderOpen } from '@fortawesome/free-solid-svg-icons';
+import { dayEnumToString } from 'common/constants/day';
+import styled from 'styled-components';
+import { CampusIcon, CellLayout } from 'client/components/general';
+
+interface NonClassMeetingsTableProps {
+  /**
+   * The list of entries to be shown in the table
+   */
+  nonClassMeetingList: NonClassMeetingResponseDTO[];
+  /**
+   * The Academic Year of the data currently being displayed
+   */
+  academicYear: number;
+}
+
+/**
+ * Utility component to style the data about a meeting
+ */
+const MeetingGrid = styled.div`
+ display: grid;
+ grid-template-areas: "time campus room";
+ grid-template-columns: 2fr 2em 3fr 2em;
+ column-gap: ${fromTheme('ws', 'xsmall')};
+ align-items: baseline;
+`;
+
+/**
+* Handles the placement of a single piece of the meeting data
+*/
+
+const MeetingGridSection = styled.div<{area: string}>`
+ grid-area: ${({ area }): string => area};
+ display: flex;
+ flex-direction: column;
+ align-items: flex-start;
+`;
```

Comment:
Good idea. I think for now it might make more sense to just import them from `tableFields.tsx`, which is what I've done in 874c7f317b9e5d3ae769d0e838f136ab1d3780e2. As we discussed in our dev check-in last week, there are plans to refactor these tables into a more generic `SemesterTable` component or something along those lines, and I would imagine part of that ticket would making the `MeetingGrid`/`MeetingGrid` section components more generic and re-usable - there's more info on this in #362 if you're curious.

Reply:
Also a287e8e894e6cc99e2a9bff365a87e0d55abdb19 for a missing import.